CMakeFuncs.cmake

```diff
@@ -0,0 +1,627 @@
+# This is a set of common functions and settings for rippled
```

Comment:
@vinniefalco AFAIK, every option was put there for a reason, and most of those reasons are probably still valid. If we want to migrate away from `scons` to `cmake`, better to support them now. Which options to remove and why is probably beyond the scope of this PR...